backends/aerospike_test.go

```diff
@@ -0,0 +1,309 @@
+	if aeKey != nil && aeKey.Value() != nil {
+		key := aeKey.Value().String()
+		c.records[key] = &as.Record{
+			Bins: binMap,
+		}
+		return nil
+	}
+	return as_types.NewAerospikeError(as_types.KEY_MISMATCH)
+}
+
+func (c *goodAerospikeClient) NewUuidKey(namespace string, key string) (*as.Key, error) {
+	return as.NewKey(namespace, setName, key)
+}
+
+func TestNewAerospikeBackend(t *testing.T) {
+	type logEntry struct {
+		msg string
+		lvl logrus.Level
+	}
+
+	testCases := []struct {
+		desc               string
+		inCfg              config.Aerospike
+		expectPanic        bool
+		expectedLogEntries []logEntry
+	}{
+		{
+			desc: "Unable to connect fakeTestUrl panic and log fatal error",
+			inCfg: config.Aerospike{
+				Host: "fakeTestUrl.foo",
+				Port: 8888,
+			},
+			expectPanic: true,
+			expectedLogEntries: []logEntry{
+				{
+					msg: "Aerospike: Failed to connect to host(s): [fakeTestUrl.foo:8888]; error: Connecting to the cluster timed out.",
+					lvl: logrus.FatalLevel,
+				},
+			},
+		},
+	}
+
+	// logrus entries will be recorded to this `hook` object so we can compare and assert them
+	hook := test.NewGlobal()
+
+	//substitute logger exit function so execution doesn't get interrupted when log.Fatalf() call comes
+	defer func() { logrus.StandardLogger().ExitFunc = nil }()
+	logrus.StandardLogger().ExitFunc = func(int) {}
+
+	for _, test := range testCases {
+		// Run test
+		assert.Panics(t, func() { NewAerospikeBackend(test.inCfg, nil) }, "Aerospike library's NewClient() should have thrown an error and didn't, hence the panic didn't happen")
+		if assert.Len(t, hook.Entries, len(test.expectedLogEntries), test.desc) {
+			for i := 0; i < len(test.expectedLogEntries); i++ {
+				assert.Equal(t, test.expectedLogEntries[i].msg, hook.Entries[i].Message, test.desc)
+				assert.Equal(t, test.expectedLogEntries[i].lvl, hook.Entries[i].Level, test.desc)
+			}
+		}
+
+		//Reset log after every test and assert successful reset
+		hook.Reset()
+		assert.Nil(t, hook.LastEntry())
+	}
+}
+
+func TestFormatAerospikeError(t *testing.T) {
+	testCases := []struct {
+		desc        string
+		inErr       error
+		inCaller    string
+		expectedErr error
+	}{
+		{
+			desc:        "Nil error",
+			inErr:       nil,
+			expectedErr: nil,
+		},
+		{
+			desc:        "Non-nil error, print without a caller",
+			inErr:       fmt.Errorf("client.Get returned nil record"),
+			expectedErr: fmt.Errorf("Aerospike: client.Get returned nil record"),
+		},
+		{
+			desc:        "Aerospike error",
+			inErr:       as_types.NewAerospikeError(as_types.SERVER_NOT_AVAILABLE),
+			inCaller:    "TEST_CASE",
+			expectedErr: fmt.Errorf("Aerospike TEST_CASE: Server is not accepting requests."),
+		},
+	}
+	for _, test := range testCases {
+		actualErr := formatAerospikeError(test.inErr, test.inCaller)
+		if test.expectedErr == nil {
+			assert.Nil(t, actualErr, "Nil error was expected")
```

Comment:
Can you please append the `desc` string at the end of the assertion message so that we know which test case the assertion failed for?